fitbenchmarking/fitbenchmarking
# Nonlinear least squares classes

#### Description of Work

Fixes #671 

Adds new nonlinear least squares cost functions and removes the `use_errors` option from docs. Together with adding the new cost function classes I have updated the results index pages so that they explicitly say what cost function is being benchmarked.

#### Testing Instructions

1. Check that there are no instances of `options.use_errors`
2. Check that the three cost functions are implemented correctly
3. Check that the analytic Jacobian is correct for the `root_nlls` option

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
